doc/modules/grid_search.rst

```diff
@@ -167,6 +171,358 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
       Random search for hyper-parameter optimization,
       The Journal of Machine Learning Research (2012)
 
+.. _successive_halving_user_guide:
+
+Searching for optimal parameters with successive halving
+========================================================
+
+Scikit-learn also provides the :class:`HalvingGridSearchCV` and
+:class:`HalvingRandomSearchCV` estimators that can be used to
+search a parameter space using successive halving [1]_ [2]_. Successive
+halving (SH) is like a tournament among candidate parameter combinations.
+SH is an iterative selection process where all candidates (the
+parameter combinations) are evaluated with a small amount of resources at
+the first iteration. Only some of these candidates are selected for the next
+iteration, which will be allocated more resources. What defines a resource is
+typically the number of samples to train on, but it can also be an arbitrary
+numeric parameter such as `n_estimators` in a random forest.
+
+As illustrated in the figure below, only a small subset of candidates
+'survive' until the last iteration. These are the candidates that have
+consistently ranked among the best candidates across all iterations. Each
+iteration is allocated an increasing amount of resources per candidate, here
+the number of samples.
+
+.. figure:: ../auto_examples/model_selection/images/sphx_glr_plot_successive_halving_iterations_001.png
+   :target: ../auto_examples/model_selection/plot_successive_halving_iterations.html
+   :align: center
+
+The ``ratio`` parameter controls the rate at which the resources grow, and
+the rate at which the number of candidates decreases. In each iteration, the
+number of resources per candidate is multiplied by ``ratio`` and the number
+of candidates is divided by the same ratio. Along with ``resource`` and
+``min_resources``, ``ratio`` is the most important parameter to control the
+search in our implementation. ``ratio`` effectively controls the number of
+iterations in :class:`HalvingGridSearchCV` and the number of candidates (if
+'auto') and iterations in :class:`HalvingRandomSearchCV`.
+``aggressive_elimination=True`` can also be used if the number of available
```

Comment:
Not sure if it makes sense to mention parameters without explaining them. Maybe just say "there are several more parameters that are explained in detail below"?

Reply:
Joel suggested having a paragraph like this one... it's hard to satisfy every reviewer when it comes to docs :p 

We do end this paragraph a few line below with

> Each parameter and their interactions are
described in more details below.

Do you think we should say that earlier?

Reply:
don't worry about it.